datafusion/core/src/execution/context/mod.rs

```diff
@@ -315,7 +320,7 @@ impl SessionContext {
     }
 
     /// Creates a new `SessionContext` using the provided [`SessionState`]
-    #[deprecated(since = "32.0.0", note = "Use SessionState::new_with_state")]
+    #[deprecated(since = "32.0.0", note = "Use SessionContext::new_with_state")]
```

Comment:
As a follow on PR perhaps we can remove this API entirely 

We could probably change the note to say say "SessionStateBuilder"